examples/MSLP_temp_winds.py

```diff
@@ -0,0 +1,155 @@
+"""
+=============================
+Model Surface Output
+=============================
+
+Plot an surface map with mean sea level pressure (MSLP),
+2m Temperature (F), and Wind Barbs (kt).
+
+"""
+
+###############################
+# Imports
+from datetime import datetime
+
+import cartopy.crs as ccrs
+import cartopy.feature as cfeature
+import matplotlib.pyplot as plt
+from metpy.units import units
+from netCDF4 import num2date
+import numpy as np
+import scipy.ndimage as ndimage
+from siphon.ncss import NCSS
+
+
+#################################
+# Helper functions
+
+# Helper function for finding proper time variable
+def find_time_var(var, time_basename='time'):
+    for coord_name in var.coordinates.split():
+        if coord_name.startswith(time_basename):
+            return coord_name
+    raise ValueError('No time variable found for ' + var.name)
+
+
+###############################################
+# Create NCSS object to access the NetcdfSubset
+# ---------------------------------------------
+# Data from NCEI GFS 0.5 deg Analysis Archive
+
+base_url = 'https://www.ncei.noaa.gov/thredds/ncss/grid/gfs-g4-anl-files/'
+dt = datetime(2018, 1, 4, 12)
+ncss = NCSS('{}{dt:%Y%m}/{dt:%Y%m%d}/gfsanl_4_{dt:%Y%m%d}_{dt:%H}00_000.grb2'.format(base_url, dt=dt))
+
+# Create lat/lon box for location you want to get data for
+query = ncss.query().time(dt)
+query.lonlat_box(north=65, south=15, east=310, west=220)
+query.accept('netcdf')
+
+# Request data for model "surface" data
+query.variables('Pressure_reduced_to_MSL_msl',
+                'Apparent_temperature_height_above_ground',
+                'u-component_of_wind_height_above_ground',
+                'v-component_of_wind_height_above_ground')
+data = ncss.get_data(query)
+
+###############################################
+# Begin data maipulation
+# -----------------------
+# Data for the surface from a model is a bit complicated.
+# The variables come from different levels and may have different data array shapes.
+#
+# MSLP: Pressure_reduced_to_MSL_msl (time, lat, lon)
+# 2m Temp: Apparent_temperature_height_above_ground (time, level, lat, lon)
+# 10m Wind: u/v-component_of_wind_height_above_ground (time, level, lat, lon)
+#
+# Height above ground Temp from GFS has one level (2m)
+# Height above ground Wind from GFS has three levels (10m, 80m, 100m)
+
+# Pull out variables you want to use
+mslp_var = data.variables['Pressure_reduced_to_MSL_msl']
```

Comment:
Could we combine this and the block below to `mslp_var = data.variables['Pressure_reduced_to_MSL_msl'][:].squeeze()`? It's a heck of a line, but reduces the total SLOC here.

Reply:
done